src/Collection.php

```diff
@@ -89,17 +89,33 @@ protected function loop() {
 	/**
 	 * This function allow to format every item that is returned to the endpoint
 	 * the filter sends 3 params to the user so can be more easy to manipulate the
-	 * data based on certaim params.
+	 * data based on certain params.
 	 *
-	 * @param \WP_Post $the_post Current post object.
 	 * @return array The formated data from every item.
 	 */
-	protected function format_item( \WP_Post $the_post ) {
-		$item = [ 'id' => $the_post->ID ];
-		if ( $this->format_item ) {
-			$item = apply_filters( Filter::ITEM_FORMAT, $item, $the_post, $this->args );
-		}
-		return $item;
+	protected function format_item() {
+		$the_post = get_post();
+
+		$the_author = get_userdata( $the_post->post_author );
+
+		$item = [
+			'id' => $the_post->ID,
+			'title' => $the_post->post_title,
+			'link' => get_permalink( $the_post->ID ),
+			'slug' => $the_post->post_name,
+			'excerpt' => get_the_excerpt(),
+			'author' => [
+				'id' => $the_author->ID,
+				'first_name' => $the_author->first_name,
+				'last_name' => $the_author->last_name,
+				'posts_link' => str_replace( home_url(), '', get_author_posts_url( $the_author->ID ) ),
```

Comment:
This field is set to autoload: http://snaps.getmoxied.net/0Z0e2C3j0w3F
This means wp will load it into memory when the app starts, there's really no performance benefit from moving this. IMO it makes the code harder to read because you have to search for where the param is defined.